activerecord/test/cases/adapters/mysql2/connection_test.rb

```diff
@@ -85,6 +85,22 @@ def test_active_after_disconnect
     assert_equal false, @connection.active?
   end
 
+  def test_wait_timeout_as_string
+    run_without_connection do |orig_connection|
+      ActiveRecord::Base.establish_connection(orig_connection.merge(wait_timeout: "60"))
+      result = ActiveRecord::Base.connection.select_value("SELECT @@SESSION.wait_timeout")
+      assert_equal 60, result
+    end
+  end
+
+  def test_wait_timeout_as_url
+    run_without_connection do |orig_connection|
+      ActiveRecord::Base.establish_connection(orig_connection.merge("url" => "mysql2:///?wait_timeout=60"))
```

Comment:
@pixeltrix Testing connection via `DATABASE_URL` is hard to me.
Actually `DATABASE_URL` is merged to `config["url"]`.

https://github.com/rails/rails/blob/0ce641839aa59d8c8306ec21cfd5f31aaa9b169c/activerecord/lib/active_record/connection_handling.rb#L76-L79

So I added a `test_wait_timeout_as_url` instead.
How about it?